web-console/src/lib/functions/felderaRelation.ts

```diff
@@ -6,17 +6,28 @@ export type CaseDependentName = {
 export const getCaseIndependentName = (entity: CaseDependentName) =>
   entity.case_sensitive ? `"${entity.name}"` : entity.name
 
+const isCaseSensitive = (caseIndependentName: string) => caseIndependentName.includes('"')
+
+/**
+ * Convert case insensitive names to lowercase, leave names in quotes as-is
+ */
+export const normalizeCaseIndependentName = (
+  caseIndependentName: string,
+  caseSensitive = isCaseSensitive(caseIndependentName)
+) => (caseSensitive ? caseIndependentName : caseIndependentName.toLocaleLowerCase())
+
 // TODO: update implementation when double quotes become a legal symbol in a name
 export const getCaseDependentName = (caseIndependentName: string) => {
+  const caseSensitive = isCaseSensitive(caseIndependentName)
   return {
```

Comment:
agree with mihai's comment that the RelationName should probably become a separate type that stores the case-sensitivity as an attribute and renders to a correct string/identifier depending on context